oak_attestation_verification/src/verifier.rs

```diff
@@ -85,15 +87,42 @@ pub fn verify_internal(
         (Some(endorsements::Type::Cb(ends)), Some(reference_values::Type::Cb(rvs))) => {
             verify_cb(evidence, ends, rvs)
         }
-        (None, None) => anyhow::bail!("Endorsements and reference values both empty"),
-        (None, Some(_)) => anyhow::bail!("Endorsements are empty"),
-        (Some(_), None) => anyhow::bail!("Reference values are empty"),
-        (Some(_), Some(_)) => anyhow::bail!("Mismatch between endorsements and reference values"),
+        (None, None) => Err(anyhow::Error::msg(
```

Comment:
Pre-existing: Looking at this again, this logic seems weird to me. Verifying the evidence would involve checking that the evidence matches either a reference value or an endorsement. This seems to check that the reference value and the endorsement matches, which would almost never happen (we will either supply an endorsement or a reference values). If neither are specified, it should skip the check. If both are specified we should either check that both match.

This can be addressed as a separate PR.

Reply:
Based on our offline discussion, I think this part is fine for now.

Reply:
Although, I think there might be use cases where we will only supply the reference values without any endorsements, but we can handle those later as needed.